Khan/wonder-blocks
# Use SearchField inside wonder blocks dropdown

## Summary:
We want to replace the current field in wb dropdown with
the newly created wb SearchField. We do this by replacing
the render content of search-text-input.js with a
`<SearchField />`. We do this instead of replacing all
of `<SearchTextInput />` entirely due to the static `isClassOf`
function and the static `_IS_SEARCH_TEXT_INPUT` variable.

- Use `SearchField` in `SearchTextInput`
- Update the tests for `SearchTextInput to RTL
- Add a test for the focus shifting to the input on clear.

Issue: WB-1281

<img width="404" alt="Screen Shot 2022-04-20 at 12 47 30 PM" src="https://user-images.githubusercontent.com/13231763/164311025-029cd867-0dbf-4fec-8ec1-3b4b42360e8c.png">


## Test plan:
`yarn jest packages/wonder-blocks-dropdown/src/components/__tests__/search-text-input.test.js`

Storybook
- Go to Dropdown/SingleSelect/Default Single Select Opened
- Click on the Filter input box
- Check there is no clear button when it's empty
- Make sure typing in it works
- Make sure clear button appears when there is text
- Use the keyboard to press the clear button
- Confirm that the focus shifts back to the input
